src/azure-cli/azure/cli/command_modules/eventgrid/_params.py

```diff
@@ -233,6 +233,7 @@ def load_arguments(self, _):    # pylint: disable=too-many-statements
 
     with self.argument_context('eventgrid topic key') as c:
         c.argument('topic_name', arg_type=name_type, help='Name of the topic', id_part=None, completer=get_resource_name_completion_list('Microsoft.EventGrid/topics'))
+        c.argument('key_name', help='Key name to regenerate key1 or key2')
```

Comment:
Why is this needed/added? Same for other key arguments for domains and partner namespaces

Reply:
never mind.. I got it.